packages/shorebird_cli/lib/src/commands/collaborators/delete_collaborators_command.dart

```diff
@@ -0,0 +1,126 @@
+import 'package:shorebird_code_push_client/shorebird_code_push_client.dart';
+
+/// {@template delete_collaborators_command}
+/// `shorebird collaborators delete`
+/// Delete an existing collaborator from a Shorebird app.
+/// {@endtemplate}
+class DeleteCollaboratorsCommand extends ShorebirdCommand
+    with AuthLoggerMixin, ShorebirdConfigMixin {
+  /// {@macro delete_collaborators_command}
+  DeleteCollaboratorsCommand({
+    required super.logger,
+    super.buildCodePushClient,
+    super.auth,
+  }) {
+    argParser
+      ..addOption(
+        _appIdOption,
+        help: 'The app id that contains the collaborator to be deleted.',
+      )
+      ..addOption(
+        _collaboratorEmailOption,
+        help: 'The email of the collaborator to delete.',
+      );
+  }
+
+  static const String _appIdOption = 'app-id';
+  static const String _collaboratorEmailOption = 'email';
+
+  @override
+  String get description =>
+      'Delete an existing collaborator from a Shorebird app.';
+
+  @override
+  String get name => 'delete';
+
+  @override
+  Future<int>? run() async {
+    if (!auth.isAuthenticated) {
+      printNeedsAuthInstructions();
+      return ExitCode.noUser.code;
+    }
+
+    final client = buildCodePushClient(
+      httpClient: auth.client,
+      hostedUri: hostedUri,
+    );
+
+    final appId = results[_appIdOption] as String? ?? getShorebirdYaml()?.appId;
+    if (appId == null) {
+      logger.err(
+        '''
+Could not find an app id.
+You must either specify an app id via the "--$_appIdOption" flag or run this command from within a directory with a valid "shorebird.yaml" file.''',
+      );
+      return ExitCode.usage.code;
+    }
+
+    final email = results[_collaboratorEmailOption] as String? ??
+        logger.prompt(
+          '''${lightGreen.wrap('?')} What is the email of the collaborator you would like to delete?''',
+        );
+
+    final getCollaboratorsProgress = logger.progress('Fetching collaborators');
+    final List<Collaborator> collaborators;
+    try {
+      collaborators = await client.getCollaborators(appId: appId);
+      getCollaboratorsProgress.complete();
+    } catch (error) {
+      getCollaboratorsProgress.fail();
+      logger.err('$error');
+      return ExitCode.software.code;
+    }
+
+    final collaborator = collaborators.firstWhereOrNull(
+      (c) => c.email == email,
+    );
+    if (collaborator == null) {
+      logger.err(
+        '''
+Could not find a collaborator with the email "$email".
+Available collaborators:
+${collaborators.map((c) => '  - ${c.email}').join('\n')}''',
+      );
+      return ExitCode.software.code;
+    }
+
+    logger.info(
+      '''
+${styleBold.wrap(lightGreen.wrap('🗑️  Ready to delete an existing collaborator!'))}
+📱 App ID: ${lightCyan.wrap(appId)}
+🤝 Collaborator: ${lightCyan.wrap(collaborator.email)}
+''',
+    );
+
+    final confirm = logger.confirm('Would you like to continue?');
```

Comment:
It might be good to include a sentence about the implications of removing a collaborator. "$email will no longer be able to publish releases or patches to this app".

Reply:
Sounds good happy to adjust the copy. I was planning to add some docs around this at docs.shorebird.dev as well.